### PR TITLE
Issue 4149 - UI - port TreeView and opther components to PF4

### PR DESCRIPTION
**Pull request by mreynolds389**
Description:  This ports all th TreeViews to PF4, and also does some proof
              of concept changes for PF3 to PF4 migration.  There is much
              more needed, but this does not break anything

relates: https://github.com/389ds/389-ds-base/issues/4149

Reviewed by: ?

### CODE REVIEW

File: src/lib389/lib389/cli_conf/backend.py [hunk 1]
```diff
@@ -348,24 +348,19 @@ def backend_get_subsuffixes(inst, basedn, log, args):
...
     if subsuf:
-        icon = "glyphicon glyphicon-leaf"
         suffix_type = "subsuffix"
     if link:
-        icon = "glyphicon glyphicon-link"
         suffix_type = "dblink"
 
     return {
-        "text": suffix,
+        "name": suffix,
```
Comment (droideck) [inline src/lib389/lib389/cli_conf/backend.py:358]: I think it may be that one healcheck test has failed because of the 'cli_conf.backend.py' change.

**Comment by droideck** [comment]
A small nitpick - the frame boundaries are weird here... I am not sure if it's an issues with PF component or it's because of the icon somehow
Screenshot from 2021-02-09 21-05-07

**Comment by droideck** [comment]
'Show Advanced Settings' line is crossed.

I see that Database tab 'Show Advanced Settings' behaves properly. Maybe we can use the same component here? (Replication tab and maybe other places too) 

Screenshot from 2021-02-09 21-06-32

**Review by droideck — CHANGES_REQUESTED** [review]
I'll check more tomorrow but that's what I found so far...

**Comment by mreynolds389** [comment]
> A small nitpick - the frame boundaries are weird here... I am not sure if it's an issues with PF component or it's because of the icon somehow
> Screenshot from 2021-02-09 21-05-07

This is chrome :-(  There is a way to disable it in our code, it has to be done in the browser itself.

File: src/lib389/lib389/cli_conf/backend.py (hunk 1 above)
Comment (mreynolds389) [reply src/lib389/lib389/cli_conf/backend.py:358]: I don't get any unexpected healthcheck errors with this PR.  Plus this code only impacts when you retrieve from the whole tree.  Healthcheck doesn't do any checks on that (not that I know of).

Comment (droideck) [reply src/lib389/lib389/cli_conf/backend.py:358]: I'm talking about the new PR CI - https://github.com/389ds/389-ds-base/pull/4604/checks?check_run_id=1865403669
But maybe there's another reason, I haven't checked manually...

Comment (mreynolds389) [reply src/lib389/lib389/cli_conf/backend.py:358]: The CI tests also pass for me when I run it locally

**Comment by mreynolds389** [comment]
> 'Show Advanced Settings' line is crossed.
> 
> I see that Database tab 'Show Advanced Settings' behaves properly. Maybe we can use the same component here? (Replication tab and maybe other places too)
> 
> Screenshot from 2021-02-09 21-06-32

I actually mean to remove the change.  I was playing around with CustomCollapse, but I ended up using a different approach and forgot to remove those code changes.  Anyway its fixed now.

**Comment by mreynolds389** [comment]
> > A small nitpick - the frame boundaries are weird here... I am not sure if it's an issues with PF component or it's because of the icon somehow
> > Screenshot from 2021-02-09 21-05-07
> 
> This is chrome :-( There is a way to disable it in our code, it has to be done in the browser itself.

Sorry I had a typo...

There is NO way to disable that in our code, it must be disabled in the Chrome browser.  A lot of people complain about this, as it was a recent change in Chrome - maybe it will get removed in a future release?

**Comment by droideck** [comment]
Okay, the second batch is on the way :)

I don't see 'Restart Instance' button. Was it moved somewhere?

Also, 'Stop Instance' fails with an error:

```
CMD: operateInstance: Do  the instance ==> dsctl -j localhost 

cockpit.js:1 usage: dsctl [-h] [-v] [-j] [-l]
             [instance]
             {restart,start,stop,status,remove,db2index,db2bak,db2ldif,dbverify,bak2db,ldif2db,backups,ldifs,tls,healthcheck,get-nsstate,ldifgen,dsrc,cockpit}
             ...
dsctl: error: invalid choice: '' (choose from 'restart', 'start', 'stop', 'status', 'remove', 'db2index', 'db2bak', 'db2ldif', 'dbverify', 'bak2db', 'ldif2db', 'backups', 'ldifs', 'tls', 'healthcheck', 'get-nsstate', 'ldifgen', 'dsrc', 'cockpit')

p @ cockpit.js:1
cockpit/$68397b70ed6a0fbee12336e132604fbdb4fa131fd183ffca6c7e8d0d6ef519ce/389-console/index.html#/:1 Uncaught SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Function.<anonymous> (index.js:111)
    at cockpit.js:1
    at cockpit.js:1
    at A (cockpit.js:1)
```

**Comment by droideck** [comment]
`Server Settings`, `Replication` and `Monitoring` - all these tabs have weird loading line alignment (Replication is not so bad but still misaligned, imo)

Screenshot from 2021-02-10 14-09-03

**Comment by droideck** [comment]
A couple of strange `Monitoring` tab errors:

1. Go to Monitoring tab;
2. Click on `Database` entry in the tree-view
And the whole tree-view is disabled now (works only if you haven't clicked on anything in the tree-view)

Another one:
1. Go to Monitoring tab;
2. Switch to `Errors Log` and enable `Continuously Refresh`
3. Switch to database suffix ('dc=example,dc=com' in my case)
Entry Cache Hit Ratio diagrams are "jumping". If `Continuously Refresh` is stopped - they are also stopped.

**Comment by mreynolds389** [comment]
@droideck - I do not see the tab issues are you facing :(  Your screen shots look nothing like mine.  

Screenshot from 2021-02-10 09-03-28

Error log continuous refresh does make the pie chart flicker, and yes somehow I dropped the "restart server" action from the dropdown.

**Comment by droideck** [comment]
> @droideck - I do not see the tab issues are you facing :( Your screen shots look nothing like mine.

I check this on Chrome with 125% zoom.
I think we better check at least Chrome and Firefox and a bit of 'moving' and 'zooming'.

But it's up to you. We can deal with that groundwork later:)

**Comment by mreynolds389** [comment]
> > @droideck - I do not see the tab issues are you facing :( Your screen shots look nothing like mine.
> 
> I check this on Chrome with 125% zoom.

If I zoom in, then I do see the misalignment with the spinner

**Comment by mreynolds389** [comment]
Okay I fixed everything you found @droideck, please review...

**Review by droideck — APPROVED** [review]
Looks great! Awesome piece of work!
Ack

**Pull request merged**